server/services/applicationService.ts

```diff
@@ -75,9 +74,22 @@ export default class ApplicationService {
     }
   }
 
-  private saveToSession(page: TasklistPage, request: Request) {
-    request.session = this.fetchOrInitializeSessionData(request.session, request.params.task, request.params.id)
-    request.session.application[request.params.id][request.params.task][request.params.page] = page.body
+  private getApplicationFromSessionOrAPI(request: Request): Promise<Application> {
+    const { application } = request.session
+
+    if (application && application.id === request.params.id) {
+      return application as unknown as Promise<Application>
```

Comment:
I feel uneasy about casting this to a `Promise` when it will be just an `Application` - I think it would be better to have a return type of `Promise<Application> | `Application`.
Also could we improve the type in the session `@types/express`?

Reply:
Good call 👍

Reply:
Fixed in https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/221/commits/61759942dc88db09739047adea3b73302245ddb3